src/main/java/org/primefaces/PrimeFaces.java

```diff
@@ -176,39 +179,51 @@ public void resetInputs(String... expressions) {
 
     /**
      * Removes the multiViewState for all DataTables within the current session.
+     *
+     * @deprecated Use {@link MultiViewState#clearMultiViewStates()} instead
      */
+    @Deprecated
     public void clearTableStates() {
-        getFacesContext().getExternalContext().getSessionMap().remove(Constants.TABLE_STATE);
+        multiViewState().clearMultiViewStates();
     }
 
     /**
      * Removes the multiViewState for one specific DataTable within the current session.
      * @param key Key of the DataTable. See {@link org.primefaces.component.datatable.DataTable#getTableState(boolean)} for the namebuild of this key.
+     *
+     * @deprecated Use {@link MultiViewState#clearMultiViewState(String, String)} instead
      */
+    @Deprecated
     public void clearTableState(String key) {
-        Map<String, Object> sessionMap = getFacesContext().getExternalContext().getSessionMap();
-        Map<String, TableState> dtState = (Map) sessionMap.get(Constants.TABLE_STATE);
-        if (dtState != null) {
-            dtState.remove(key);
-        }
+        clearDataListState(key);
     }
 
     /**
      * Removes the multiViewState for all DataLists within the current session.
+     *
+     * @deprecated Use {@link MultiViewState#clearMultiViewStates()} instead
      */
+    @Deprecated
     public void clearDataListStates() {
-        getFacesContext().getExternalContext().getSessionMap().remove(Constants.DATALIST_STATE);
+        multiViewState().clearMultiViewStates();
     }
 
     /**
      * Removes the multiViewState for one specific DataList within the current session.
      * @param key Key of the DataList. See {@link org.primefaces.component.datalist.DataList#getDataListState(boolean)}} for the namebuild of this key.
+     *
+     * @deprecated Use {@link MultiViewState#clearMultiViewState(String, String)} instead
      */
+    @Deprecated
     public void clearDataListState(String key) {
-        Map<String, Object> sessionMap = getFacesContext().getExternalContext().getSessionMap();
-        Map<String, TableState> dtState = (Map) sessionMap.get(Constants.DATALIST_STATE);
-        if (dtState != null) {
-            dtState.remove(key);
+        Pattern p = Pattern.compile("(.*\\.(?:xhtml|jsf|jsp))_(.*)");
```

Comment:
I like to move Patterns to private static final constant so that way the pattern is compiled only once which is the expensive part.

Reply:
I like to do that too, but I can't declare a static variable in inner class (though the compiler seems to be if it's a String, I didn't find an explanation for that). Since this method is deprecated and should be removed in the future, I didn't want to spend too much time on it

Reply:
No problem!  Approving PR.